### PR TITLE
Added krunner-symbols

### PR DESCRIPTION
**Pull request by hqurve**
###### Motivation for this change
I would like to use the addon as it is very convenient. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hqurve** [comment]
I'm not sure how to label this, but I tested the package in a virtual machine by adding it to systemPackages and calling with `libsForQt5.callPackage`.

**Comment by urlordjames** [comment]
See [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), you need to add yourself to `maintainers/maintainer-list.nix` (in a separate commit) then add yourself to `meta.maintainers`.

File: pkgs/desktops/plasma-5/3rdparty/addons/krunner-symbols.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    sed -i 's,LOCATION_PLUGIN,KDE_INSTALL_PLUGINDIR,' CMakeLists.txt
+    sed -i 's,LOCATION_DESKTOP,KDE_INSTALL_KSERVICES5DIR,' CMakeLists.txt
+  '';
+
+  cmakeFlags = [ "-DLOCATION_CONFIG=share/config" ];
+
+  meta = with lib; {
+    description = "A little krunner plugin (Plasma 5) to retrieve unicode symbols, or any other string, based on a corresponding keyword";
+    homepage = "https://github.com/domschrei/krunner-symbols";
+    license = licenses.gpl3;
```
Comment (urlordjames) [inline pkgs/desktops/plasma-5/3rdparty/addons/krunner-symbols.nix:31]: Fairly sure `licenses.gpl3` is deprecated, I believe you're supposed to use `licenses.gpl3Only` or `licenses.gpl3Plus` now.

File: pkgs/desktops/plasma-5/3rdparty/addons/krunner-symbols.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    owner = "domschrei";
+    repo = "krunner-symbols";
+    rev = version;
+    sha256 = "sha256-YsoZdPTWpk3/YERwerrVEcaf2IfGVJwpq32onhP8Exo=";
+  };
+
+  buildInputs = [ qtbase ki18n kdelibs4support krunner ];
+  nativeBuildInputs = [ cmake wrapQtAppsHook extra-cmake-modules ];
+
+  prePatch = ''
```
Comment (urlordjames) [inline pkgs/desktops/plasma-5/3rdparty/addons/krunner-symbols.nix:19]: I would leave some comment explaining the purpose of this.

**Review by urlordjames — COMMENTED** [review]
I would also squash these commits down to two (one adding yourself to the maintainer list and the other with all the rest of the changes).

File: pkgs/desktops/plasma-5/3rdparty/addons/krunner-symbols.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
```
Comment (SuperSandro2000) [inline pkgs/desktops/plasma-5/3rdparty/addons/krunner-symbols.nix:19]:
```suggestion
  postPatch = ''
```

File: pkgs/desktops/plasma-5/3rdparty/addons/krunner-symbols.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    # symbols.cpp hardcodes the location of configuration files
+    sed -i 's,/usr/share/config/krunner-symbol,'$out'/share/config/krunner-symbol,' symbols.cpp
+
+    # change cmake flag names to output using the correct qt-plugin prefix
+    sed -i 's,LOCATION_PLUGIN,KDE_INSTALL_PLUGINDIR,' CMakeLists.txt
+
+    # change cmake flag names to output using the correct kservice location
+    sed -i 's,LOCATION_DESKTOP,KDE_INSTALL_KSERVICES5DIR,' CMakeLists.txt
```
Comment (SuperSandro2000) [inline pkgs/desktops/plasma-5/3rdparty/addons/krunner-symbols.nix:27]:
```suggestion
    # change cmake flag names to output using the correct qt-plugin prefix and  kservice location
    substituteInPlace CMakeLists.txt \
      --replace "LOCATION_DESKTOP" "KDE_INSTALL_KSERVICES5DIR" \
      --replace "LOCATION_PLUGIN" "KDE_INSTALL_PLUGINDIR"
```

File: pkgs/desktops/plasma-5/3rdparty/addons/krunner-symbols.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    rev = version;
+    sha256 = "sha256-YsoZdPTWpk3/YERwerrVEcaf2IfGVJwpq32onhP8Exo=";
+  };
+
+  buildInputs = [ qtbase ki18n kdelibs4support krunner ];
+  nativeBuildInputs = [ cmake wrapQtAppsHook extra-cmake-modules ];
+
+  prePatch = ''
+    # symbols.cpp hardcodes the location of configuration files
+    sed -i 's,/usr/share/config/krunner-symbol,'$out'/share/config/krunner-symbol,' symbols.cpp
```
Comment (SuperSandro2000) [inline pkgs/desktops/plasma-5/3rdparty/addons/krunner-symbols.nix:21]:
```suggestion
    substituteInPlace symbols.cpp \
      --replace "/usr/share/config/krunner-symbol" "$out'/share/config/krunner-symbol"
```

File: pkgs/desktops/plasma-5/3rdparty/addons/krunner-symbols.nix [hunk 6]
```diff
@@ -0,0 +1,39 @@
+{ lib, stdenv
+, cmake, fetchFromGitHub, extra-cmake-modules
+, qtbase, wrapQtAppsHook, ki18n, kdelibs4support, krunner
+}:
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/desktops/plasma-5/3rdparty/addons/krunner-symbols.nix:5]:
```suggestion
}:

stdenv.mkDerivation rec {
```

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -4285,6 +4285,12 @@
     githubId = 6074754;
     name = "Hlodver Sigurdsson";
   };
+  hqurve = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:4288]: Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME`` into this PR.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by aanderse** [comment]
@hqurve is there anything special I have to do to be able to use this? I added this to my `environment.systemPackages` but the plugin never showed up in `krunner`.

**Comment by hqurve** [comment]
Yes you just add it to `environment.systemPackages` and it should appear (I dont recall if a restart is necessary)
Screenshot from 2022-02-18 06-26-10

In fact I have it added through home-manager using `home.packages` and it also works,

**Comment by aanderse** [comment]
Something must have been funky with my desktop... it only worked after logging out and then logging back in. Thanks!

**Pull request merged**